### PR TITLE
Add local fallback

### PR DESCRIPTION
**Pull request by seekayel**
Tried to add local fs fallback for local testing and compatibility.

`readFile()` needs a different signature and call pattern (aka CB vs Promise)

Also, I'm not in love with how `this.isLocal` is calculated.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -15,13 +18,27 @@ function streamToBuffer(stream) {
 }
 
 class CyclicS3FS {
-  constructor(bucketName, config={}) {
+  constructor(bucketName='', config={}) {
     this.bucket = bucketName
     this.config = config
     this.s3 = new S3Client({...config});
+    this.isLocal = !bucketName
```
Comment (korostelevm) [inline src/index.js:25]: should do this based on no AWS creds I think 
and/or another flag in the config object 

in which case probably should be more explicit about what config goes into instantiating the s3 client 
AWS region, etc

File: src/index.js [hunk 2]
```diff
@@ -33,25 +50,31 @@ class CyclicS3FS {
...
+      return fs.writeFile(fileName, data, options)
+    }
     const cmd = new PutObjectCommand({
         Bucket: this.bucket,
         Key: fileName,
         Body: data
     })
 
-    await this.s3.send(cmd)
+    return await this.s3.send(cmd)
```
Comment (korostelevm) [inline src/index.js:62]: write file is supposed to return undefined

File: src/index.js [hunk 3]
```diff
@@ -33,25 +50,31 @@ class CyclicS3FS {
   }
 
   async writeFile(fileName, data, options={}) {
+    if (this.isLocal){
```
Comment (korostelevm) [inline src/index.js:53]: instead of doing all these ifs everywhere, just return regular fs from the constructor right after the check

Comment (korostelevm) [reply src/index.js:53]: can't really do it in the constructor actually but maybe a static method/factory

Comment (seekayel) [reply src/index.js:53]: 👍

**Comment by seekayel** [comment]
Superseded by other PRs. Closing.

**Pull request closed without merging**